plugins/org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/parser/antlr/XtextAntlrGeneratorComparisonFragment.xtend

```diff
@@ -0,0 +1,318 @@
+				handleMismatch(absoluteGrammarFileName, absoluteGrammarFileNameReference,
+					match, matchReference, newlineCounter, newlineCounterReference
+				)
+			}
+			
+			continue = continue && compoundMatcherReference.find && compoundMatcher.find
+		}
+		
+		return Pair.of(newlineCounter, newlineCounterReference)
+	}
+	
+	/**
+	 * continue matching until the next token is found, count newlines while doing so.
+	 * 
+	 * @return the number of newlines passed while searching 
+	 */
+	def private nextToken(Matcher matcher, int previousEnd, ()=> void missedCharSequenceHandler) {
+		var continue = true;
+		var newlineCounter = 0;
+		var thePreviousEnd = previousEnd
+		
+		while (continue) {
+			if (matcher.start() != thePreviousEnd) {
+				missedCharSequenceHandler.apply()
+			}
+			
+			val match = matcher.group()
+			
+			if (p_newline.matcher(match).matches()) {
+				// in case the 'newline' pattern matches the found 'match' ...
+				newlineCounter++
+				thePreviousEnd = matcher.end
+				
+			} else if (p_ws.matcher(match).matches) {
+				thePreviousEnd = matcher.end
+				
+			} else if (p_token.matcher(match).matches()) {
+				// in case a valid token has been found stop here
+				return newlineCounter
+			}
+			
+			continue = matcher.find()
+		}
+		
+		return newlineCounter
+	}
+
+	protected def handleUnexpectedCharSequence(String absoluteGrammarFileName, int lineNo) {
+		deleteDir(tmpDir)
+		
+		throw new RuntimeException('''
+			Noticed an unmatched character sequence in file �absoluteGrammarFileName� in/before line �lineNo�.
+		''')
+	}
+
+	protected def handleMismatch(String absoluteGrammarFileName, String absoluteGrammarFileNameReference,
+			String match, String matchReference, int lineNo, int lineNoReference) {
+		copyFile(absoluteGrammarFileNameReference, absoluteGrammarFileName)
+		
+		deleteDir(tmpDir)
+		
+		throw new RuntimeException('''
+			Generated grammar �absoluteGrammarFileName� differs at token �match� (line �lineNo
+				�), expected token �matchReference� (line �lineNoReference�).''')
+	}
+	
+	protected def copyFile(String from, String to) {
+		Files.copy(
+			new File(from),
+			new File('''�to.substring(0, to.length - 2)�Expected.g''')
+		)
+	}
+
+	/**
+	 * offers a singleton temporary folder 
+	 */
+	private def File create path: Files.createTempDir() getTmpDir() {
+	}
+
+	/**
+	 * Tweaks the generation of the {@link Generator#SRC_GEN Generator.SRC_GEN} outlet
+	 * and injects the {@link #getTmpPath()}.
+	 */
+	override protected createOutlet(boolean append, String encoding, String name, boolean overwrite, String path) {
+		if (name == Generator.SRC_GEN || name == Generator.SRC_GEN_IDE || name == Generator.SRC_GEN_UI) {
+			super.createOutlet(append, encoding, name, overwrite, getTmpDir().absolutePath)			
+		} else {			
+			super.createOutlet(append, encoding, name, overwrite, path)
+		}
+	}
+
+	/** little helper for cleaning up the temporary stuff. */
+    private static def void deleteDir(File dir) {
```

Comment:
You may want to use org.eclipse.xtext.util.Files.sweepFolder(File) instead to delete all the children.